addons/mail/controllers/discuss/channel.py

```diff
@@ -54,6 +56,41 @@ def discuss_channel_pins(self, channel_id):
             raise NotFound()
         return channel.pinned_message_ids.sorted(key="pinned_at", reverse=True).message_format()
 
+    @http.route("/discuss/channel/mute", methods=["POST"], type="json", auth="user")
+    def discuss_channel_mute(self, channel_id, minutes):
+        """Mute notifications for the given number of minutes.
+        :param minutes: (integer) number of minutes to mute notifications, -1 means mute until the user unmutes
+        """
+        member = request.env["discuss.channel.member"].search([("channel_id", "=", channel_id), ("is_self", "=", True)])
+        if not member:
+            raise request.not_found()
+        if minutes == -1:
+            member.mute_until_dt = datetime.max
+        elif minutes:
+            member.mute_until_dt = fields.Datetime.now() + relativedelta(minutes=minutes)
+            request.env.ref("mail.ir_cron_discuss_channel_member_unmute")._trigger(member.mute_until_dt)
+        else:
+            member.mute_until_dt = False
+        channel_data = {
+            "id": member.channel_id.id,
+            "model": "discuss.channel",
+            "mute_until_dt": member.mute_until_dt,
+        }
+        request.env["bus.bus"]._sendone((member.partner_id, "mail.record/insert", {"Thread": channel_data}))
```

Comment:
review note: sending on bus for all changes so that all tabs/devices are sync with new values